beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -0,0 +1,170 @@
+		}
+	}
+	http2.WriteJson(w, &ListAttestationsResponse{Data: filteredAtts})
+}
+
+// SubmitAttestations submits an Attestation object to node. If the attestation passes all validation
+// constraints, node MUST publish the attestation on an appropriate subnet.
+func (s *Server) SubmitAttestations(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.SubmitAttestations")
+	defer span.End()
+
+	if r.Body == http.NoBody {
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
+	var req SubmitAttestationsRequest
+	if err := json.NewDecoder(r.Body).Decode(&req.Data); err != nil {
+		http2.HandleError(w, "Could not decode request body: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+	if len(req.Data) == 0 {
+		http2.HandleError(w, "No data submitted", http.StatusBadRequest)
+		return
+	}
+	validate := validator.New()
+	if err := validate.Struct(req); err != nil {
+		http2.HandleError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var validAttestations []*ethpbalpha.Attestation
+	var attFailures []*shared.IndexedVerificationFailure
+	for i, sourceAtt := range req.Data {
+		att, err := sourceAtt.ToConsensus()
+		if err != nil {
+			attFailures = append(attFailures, &shared.IndexedVerificationFailure{
+				Index:   i,
+				Message: "Could not convert request attestation to consensus attestation: " + err.Error(),
+			})
+			continue
+		}
+		if _, err = bls.SignatureFromBytes(att.Signature); err != nil {
+			attFailures = append(attFailures, &shared.IndexedVerificationFailure{
+				Index:   i,
+				Message: "Incorrect attestation signature: " + err.Error(),
+			})
+			continue
+		}
+
+		// Broadcast the unaggregated attestation on a feed to notify other services in the beacon node
+		// of a received unaggregated attestation.
+		// Note we can't send for aggregated att because we don't have selection proof.
+		if !corehelpers.IsAggregated(att) {
+			s.OperationNotifier.OperationFeed().Send(&feed.Event{
+				Type: operation.UnaggregatedAttReceived,
+				Data: &operation.UnAggregatedAttReceivedData{
+					Attestation: att,
+				},
+			})
+		}
+
+		validAttestations = append(validAttestations, att)
+	}
+
+	failedBroadcasts := make([]string, 0)
+	for i, att := range validAttestations {
+		// Determine subnet to broadcast attestation to
+		wantedEpoch := slots.ToEpoch(att.Data.Slot)
+		vals, err := s.HeadFetcher.HeadValidatorsIndices(ctx, wantedEpoch)
+		if err != nil {
+			http2.HandleError(w, "Could not get head validator indices: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		subnet := corehelpers.ComputeSubnetFromCommitteeAndSlot(uint64(len(vals)), att.Data.CommitteeIndex, att.Data.Slot)
+
+		if err = s.Broadcaster.BroadcastAttestation(ctx, subnet, att); err != nil {
+			failedBroadcasts = append(failedBroadcasts, strconv.Itoa(i))
+			log.WithError(err).Error("could not broadcast attestation")
```

Comment:
maybe we include i here so you can find which index it's related to. maybe associate with attFailures